runtime/src/mocks.rs

```diff
@@ -37,7 +40,11 @@ impl MockStream {
 }
 
 impl Stream for MockStream {
-    async fn recv(&mut self) -> Result<Bytes, Error> {
-        self.receiver.next().await.ok_or(Error::ReadFailed)
+    async fn recv(&mut self, buf: &mut [u8]) -> Result<(), Error> {
+        let msg = self.receiver.next()
+            .await
+            .ok_or(Error::ReadFailed)?;
+        buf.copy_from_slice(&msg);
```

Comment:
In the current implementation, there is no guarantee this will work (we should be reading `buf.len()` bytes from self.receiver